internal/function/commit_has_blob.go

```diff
@@ -14,13 +16,18 @@ import (
 type CommitHasBlob struct {
 	commitHash sql.Expression
 	blob       sql.Expression
+	cache      *lru.ARCCache
 }
 
+const commitHasBlobCacheSize = 200
+
 // NewCommitHasBlob creates a new commit_has_blob function.
 func NewCommitHasBlob(commitHash, blob sql.Expression) sql.Expression {
+	cache, _ := lru.NewARC(commitHasBlobCacheSize)
```

Comment:
Maybe it's a silly question but, isn't there any problem using `ARC cache` given that IBM has the patent of it?

Reply:
Actually, not silly at all. I think we should move it to `TwoQueueCache`, then. What do you think? @ajnavarro @jfontan